### DISCUSSION TITLE
Duplicate the nested documents together with its parent document.

### DISCUSSION THREAD

**Discussion by DestinyJazz** [outline/outline#5332] [Ideas]
It would be handy if we could have the option to duplicate the nested documents together for easier templating. It is quite troublesome when we need to duplicate all the nested documents individually and choose where to move.

**Comment by tommoor** [comment]
@DestinyJazz seems reasonable, what is your usecase for duplicating a tree of documents in this case?

**Reply by DestinyJazz** [reply]
I am using it to manage product manual and guide. Therefore when newer versions released, i would like to update the few sections that cover the latest update while the rest of the contents in nested documents remain the same.

Duplicate the nested docuements together would be helpful.

**Reply by tommoor** [reply]
Oh interesting, so there is a lot of duplicated content between versions within Outline then?

**Reply by DestinyJazz** [reply]
Eventually, yes. 
Should I create an issue based on what I suggested?

**Reply by margsenjo** [reply]
@tommoor for us it would be awesome as well. We store client information in outline and each new client gets their own page tree. In this case would be super super helpful.

**Reply by sncat** [reply]
This is a great feature and I would love to see it implemented.
It would be very useful in cases where we have the same document hierarchy, but with different content information (users, IPs, etc.).

**Reply by tommoor** [reply]
Agreed it would be good, but we have to move duplication to the backend API to achieve this – currently it's just a client operation, so that makes it more complicated

**Reply by sncat** [reply]
I now understand that this is not an easy implementation.😢

It is a point of concern that when a document is copied, nested child documents are not copied at the same time.
(I think this is due to the fact that we think in terms of the directory structure of Windows, etc.)

It does not have to be right now, and I would be very grateful if you would consider implementing this in the future.🙏

**Reply by tommoor** [reply]
This already landed actually, https://github.com/outline/outline/pull/6009 – it is live on cloud hosted and will be in the next OSS release in about 3 weeks.
